### PR TITLE
Specify Span ID creation with sampling (non-recording spans included)

### LINKED ISSUE #1060: Missing spec for Span ID creation for dropped/record-only spans
**What are you trying to achieve?**

I am trying to ensure that span ID creation for spans that are dropped or only recorded (but not sampled) is uniform across different language implementations.

**Additional context.**

As explained in https://github.com/open-telemetry/opentelemetry-specification/issues/864#issuecomment-689513021 the spec is silent on what to do for IDs of spans that are not sampled. #998 suggest to reuse the parent-span-id for dropped or use zero if there is none, but use a new span ID for everything else. A simpler alternative would be to simply add the sentence "A new Span ID MUST always be generated, even for dropped spans".

This is not purely an optimization issue, it impacts the observable behavior of the monitored application when the unsampled span is injected (and then potentially used as parent on the other side).

### PR DESCRIPTION
**Pull request by carlosalberto**
Fixes #1060 

Spin-off of #998 (closes #998)

After discussing pros and cons, [it was recommended](https://github.com/open-telemetry/opentelemetry-specification/pull/998/files#r517616726) to always generate new Span Ids even for non-recording instances.

> The TC recommendation is to generate a new span ID even for non-recording spans.

cc @yurishkuro @SergeyKanzhelev

### CODE REVIEW

File: spec-compliance-matrix.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ status of the feature is not known.
 |[Sampling](specification/trace/sdk.md#sampling)|
 |Allow samplers to modify tracestate           |   | +  |   | [-](https://github.com/open-telemetry/opentelemetry-python/issues/1220)    |    | +    |   | +  |   |    |
 |ShouldSample gets full parent Context         |   | +  | + | +    |    | +    |   |    |   |    |
+|[Span/Trace ID creation wrt. sampling is compliant](specification/trace/sdk.md#sdk-span-creation) |   |    |   |      |    |      |   |    |   |    |
```
Comment (Oberon00) [inline spec-compliance-matrix.md:71]:
```suggestion
|[New Span ID created also for non-recording Spans](specification/trace/sdk.md#sdk-span-creation) |   |    |   |      |    |      |   |    |   |    |
```

**Review by Oberon00 — COMMENTED** [review]
Comparison to the alternative PR #998: https://github.com/dynatrace-oss-contrib/opentelemetry-specification/compare/sdk-spanid-sampling...carlosalberto:sdk-spanid-sampling

File: spec-compliance-matrix.md (hunk 1 above)
Comment (carlosalberto) [reply spec-compliance-matrix.md]: Thanks! Applied.

**Comment by Oberon00** [comment]
One point to consider: If we explicitly encourage users to rely on unsampled spans having an ID, we should probably close #864 "Option to allow "default" IDs for unsampled traces" as wontfix. CC @awssandra

**Review by bogdandrutu — APPROVED** [review]

File: specification/trace/sdk.md [hunk 2]
```diff
@@ -61,11 +61,28 @@ The following table summarizes the expected behavior for each combination of
...
+3. Generate a new span ID for the `Span`, independently of the sampling decision.
+   This is done so other components (such as logs or exception handling) can rely on
+   a unique span ID, even if the `Span` is a non-recording instance.
+4. Create a span depending on the decision returned by `ShouldSample`:
+   see description of [`ShouldSample`'s](#shouldsample) return value below
+   for how to set `IsRecording` and `Sampled` on the Span,
+   and the [table above](#recording-sampled-reaction-table) on whether
+   to pass the `Span` to `SpanProcessor`s.
+   A non-recording span MAY be implemented using the same mechanism as when a
+   `Span` is created with no API-implementation installed or as described in
```
Comment (jkwatson) [inline specification/trace/sdk.md:84]:
```suggestion
   `Span` is created without an SDK installed or as described in
```

Comment (carlosalberto) [reply specification/trace/sdk.md]: Thanks!

**Review by yurishkuro — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Pull request merged**